river/tree/nodes/hatr_nodes.py

```diff
@@ -2,8 +2,7 @@
 import random
 import typing
 
-from river.drift.adwin import ADWIN
-from river.stats import Var
+from river.stats import Mean, Var
```

Comment:
As a general rule I would encourage `from river import stats`

Reply:
Me too. However, ironically and historically, we have `stats` as an init-param of the nodes. So, when I try `stats.Mean,` the issued dictionary parameter is accessed rather than the module. I might do something like: `from river import stats as sts`.

Currently, I am scanning every possible paper to try to understand this line:

https://github.com/online-ml/river/blob/46f8e874e328d7f37e8461eb700a428fb98ab1a7/river/tree/nodes/hatc_nodes.py#L277-L279

This portion is the core aspect of the algorithm that I can't find in the available formulae. The idea of resetting the detectors after each drift breaks the conditional on replacing the alternate trees. So I am trying to refactor the tree and make it more generic.

This mysterious bound is on my way :P